crates/red_knot/src/types/infer.rs

```diff
@@ -95,6 +82,13 @@ where
                 ty
             }
         }
+        Definition::Assignment(node_key) => {
+            let parsed = db.parse(file_id);
+            let ast = parsed.ast();
+            let node = node_key.resolve_unwrap(ast.as_any_node_ref());
```

Comment:
I can see how resolving definitions might become expensive :laughing:

Reply:
Yes, it would be awesome if we could directly reference AST nodes instead of having to search to resolve :/